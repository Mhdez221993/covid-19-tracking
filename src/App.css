* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  text-decoration: none;
}

body {
  background-color: #fc4f91;
  color: #f8e0e9;
  font-family: Arial, Helvetica, sans-serif;
}

img {
  width: 100%;
}

.navBar {
  display: flex;
  align-items: center;
  padding: 20px;
  background-color: #d14278;
}

.all-cases {
  margin: 0 auto;
  font-size: 20px;
  font-weight: 100;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.setting {
  margin-left: 20px;
}

.back {
  display: flex;
  align-items: center;
  justify-content: center;
}

.back-arrow {
  font-size: 20px;
  color: inherit;
  color: #f8e0e9;
}

.euro-wrapper {
  display: flex;
}

.euro-map {
  flex: 1;
}

.europ-description {
  flex: 1;
  font-size: 20px;
  padding-left: 30px;
}

.country-item {
  min-height: 140px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-end;
  padding: 10px 10px 0 0;
  background-color: #df4782;
}

li.country-item:nth-child(4n+2),
li.country-item:nth-child(4n+3) {
  background-color: #d14278;
}

.country-wrapper {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: auto;
}

.arrow-right {
  font-size: 22px;
  color: #f8e0e9;
}

.title-country {
  font-weight: 700;
  font-size: 18px;
}

.p-euro {
  margin-top: 50px;
  font-size: 25px;
}

.city-wrapper {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto;
}

.city-item {
  min-height: 90px;
  display: grid;
  grid-template-columns: 1fr 80px 30px;
  padding: 10px;
  align-items: center;
}

li.city-item:nth-child(odd) {
  background-color: #d14278;
}

li.city-item:nth-child(even) {
  background-color: #df4782;
}

.city-number {
  margin: 0 auto;
}

.city-section {
  width: 100%;
  text-align: center;
  padding: 10px 0;
  background-color: #df4782;
}
