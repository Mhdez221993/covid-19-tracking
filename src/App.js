import React, { useState } from 'react';
import {
  Switch,
  Route,
} from 'react-router-dom';
import City from './components/City';
import Country from './components/Country';
import Nav from './components/Nav';

function App() {
  const [cities, setCities] = useState([]);
  const [country, setCityName] = useState('');
  const handleCity = (country, categories) => {
    setCities(categories);
    setCityName(country);
  };

  return (
    <div className="App">
      <div className="main">
        <Nav />
        <Switch>
          <Route exact path="/">
            <Country handleCity={handleCity} />
          </Route>
          <Route exact path="/city">
            <City cities={cities} country={country} />
          </Route>
        </Switch>
      </div>
    </div>
  );
}

export default App;
