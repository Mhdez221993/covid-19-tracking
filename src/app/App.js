import React from 'react';
import {
  BrowserRouter as Router, Route, Routes,
} from 'react-router-dom';
import Cities from '../features/cities/Cities';
import Countries from '../features/countries/Countries';
import Nav from '../features/Nav/Nav';

const routes = [
  {
    path: '/',
    name: 'Countries',
    component: <Countries />,
  },
  {
    path: '/country/:name',
    name: 'Cities',
    component: <Cities />,
  },
];

const App = () => (
  <Router>
    <Nav />
    <Routes>
      {routes.map(({ path, component }) => (
        <Route key={path} path={path} element={component} />
      ))}
    </Routes>
  </Router>
);

export default App;
