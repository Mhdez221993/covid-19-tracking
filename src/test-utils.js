import { render as rtlRender } from '@testing-library/react';
import PropTypes from 'prop-types';
import React from 'react';
import { Provider } from 'react-redux';
import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import continent from './features/continent/continentReducer';
import country from './features/country/countryReducer';

function createTestStore() {
  const store = createStore(
    combineReducers({
      continent,
      country,
    }),
    applyMiddleware(thunk),
  );
  return store;
}

const reducer = combineReducers({
  continent,
  country,
});

const store = createStore(
  reducer,
  applyMiddleware(thunk),
);

function testRender(
  ui,
  {
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  Wrapper.propTypes = {
    children: PropTypes.node.isRequired,
  };

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { testRender, createTestStore };
