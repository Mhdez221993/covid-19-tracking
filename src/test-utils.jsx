import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import countries from './redux/countries/country';

function render(
  ui,
  {
    preloadedState,
    store = configureStore({
      countries,
      preloadedState,
    }),
    ...renderOptions
  } = {},
) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <Router>{children}</Router>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
