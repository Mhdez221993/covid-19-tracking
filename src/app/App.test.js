import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createTestStore, testRender } from '../test-utils';
import App from './App';

let store;
describe('App Component with API call', () => {
  beforeEach(() => {
    store = createTestStore();
  });
  test('Your component with a full reducer flow', async () => {
    // Create a redux store
    const { findByText } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    await findByText('Spain');
  });

  test('App snapshot test', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('App component without API call', () => {
  test('Shour render Spain not in the document', () => {
    testRender(<App />);
    const linkElement = screen.queryByText('Spain');
    expect(linkElement).not.toBe();
  });
});
