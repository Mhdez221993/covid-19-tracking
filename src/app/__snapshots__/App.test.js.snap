// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component with API call App snapshot test 1`] = `
Array [
  <nav
    className="navBar"
  >
    <div
      className="back"
    >
      <a
        exact="true"
        href="/"
        onClick={[Function]}
      >
        <svg
          className="back-arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </a>
      <span
        className="date"
      >
        2021
      </span>
    </div>
    <span
      className="all-cases"
    >
      Cases in Europe
    </span>
    <div
      className="tools"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
        />
        <path
          d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
        />
      </svg>
      <svg
        className="setting"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="3"
        />
        <path
          d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
        />
      </svg>
    </div>
  </nav>,
  <div
    className="wrapper"
  >
    <div
      className="euro-wrapper"
    >
      <div
        className="euro-map"
      >
        <img
          alt="europe"
          src="europe.png"
        />
      </div>
      <div
        className="europ-description"
      >
        <p
          className="p-euro title-country"
        >
          EUROPE
        </p>
        0
      </div>
    </div>
    <ul
      className="country-wrapper"
    />
  </div>,
]
`;
